ext/include/opentelemetry/ext/zpages/zpages_http_server.h

```diff
@@ -0,0 +1,188 @@
+   */
+  std::string GetQuery(const std::string& uri) {
+    if (endpoint_.length() + 1 > uri.length()) return uri;
+    return uri.substr(endpoint_.length() + 1);
+  }
+
+  /*
+   * Helper that returns whether a str starts with pre
+   * @param str is the string we're checking
+   * @param pre is the prefix we're checking against
+   */
+  bool StartsWith(const std::string& str, std::string pre) {
+    return (pre.length() > str.length())
+        ? false
+        : str.substr(0, pre.length()) == pre;
+  }
+
+  /*
+   * Helper that returns the remaining string after the leftmost backslash
+   * @param str is the string we're extracting from
+   */
+  std::string GetAfterSlash(const std::string& str) {
+    const auto& backslash = str.find("/");
+    if (backslash == std::string::npos || backslash == str.length()) return "";
+    return str.substr(backslash + 1);
+  }
+
+  /*
+   * Helper that returns the remaining string after the leftmost backslash
+   * @param str is the string we're extracting from
+   */
+  std::string GetBeforeSlash(const std::string& str) {
+    const auto& backslash = str.find("/");
+    if (backslash == std::string::npos || backslash == str.length()) return str;
+    return str.substr(0, backslash);
+  }
+  
+  const std::string endpoint_;
+
+ private:
+  /*
+   * Return whether a file is found whose location is searched for relative to where
+   * the executable was triggered. If the file is valid, fill result with the file data/information
+   * required to display it on a webpage
+   * @param name of the file to look for,
+   * @param resulting file information, necessary for displaying them on a webpage
+   * @returns whether a file was found and result filled with display information
+   */
+  bool FileGetSuccess (std::string filename, std::vector<char>& result) {
+    #ifdef _WIN32
+    std::replace(filename.begin(), filename.end(), '/', '\\');
+    #endif
+    std::streampos size;
+    std::ifstream file(filename, std::ios::in | std::ios::binary | std::ios::ate);
+    if (file.is_open()) {
+      size = file.tellg();
+      if (size) {
+          result.resize(size);
+          file.seekg(0, std::ios::beg);
+          file.read(result.data(), size);
+      }
+      file.close();
+      return true;
+    }
+    return false;
+  };
+
+  /*
+   * Returns the extension of a file
+   * @param name of the file
+   * @returns file extension type under HTTP protocol
+   */
+  std::string GetMimeContentType(std::string filename) {
+    std::string file_ext = filename.substr(filename.find_last_of(".") + 1);
+    auto file_type = mime_types_.find(file_ext);
+    return (file_type != mime_types_.end())
+        ? file_type->second
+        : HTTP_SERVER_NS::CONTENT_TYPE_TEXT;
+  };
+
+  /*
+   * Returns the standardized name of a file by removing backslashes, and
+   * assuming index.html is the wanted file if a directory is given
+   * @param name of the file
+   */
+  std::string GetFileName(std::string S) {
```

Comment:
The method shouldn't change the input argument.

```suggestion
  std::string GetFileName(const std::string& S) {
  std::string file_name; 
  ...
  }
```

Reply:
For this specific function, could I keep it as is? The original string is const, and I want to return a modified version of it depending on its contents

Reply:
If you want to have a method that is changing the input, you can do:

void AdjustFileName(std::string& str) {...}

'void' because we don't return anything new, instead we change the contents of |str|. I would also rename it to be something else and not Get<>,  that reflects that the string is changed. Same for the comment, you can write maybe: " Modifies/updates |str| to be the standardized name of a file by removing backslashes...".

(And nit: the input variable should be lower case).

Reply:
I tried this and it doesn't work because the original input string is type const string (from req.uri, and req is type HttpRequest const&). The current signature makes a modifiable copy of the input to send. I'm not sure if this is the most idiomatic approach, but that's the best working solution I have so far? Using `std::string GetFileName(const std::string& name)` would mean I have to still make a copy of the input to send, and `std::string&` isn't allowed because it's const